build_config/lib/src/build_config.dart

```diff
@@ -179,18 +182,59 @@ class BuildTarget {
   /// Whether or not this is the default dart library for the package.
   final bool isDefault;
 
-  /// Sources to use as inputs for `builders`. May be `null`, in which case
-  /// it should fall back on `sources`.
-  final Iterable<String> generateFor;
-
   BuildTarget(
       {this.builders: const {},
       this.dependencies,
       this.platforms: const [],
       this.excludeSources: const [],
-      this.generateFor,
       this.isDefault: false,
       this.name,
       this.package,
       this.sources: const ['lib/**']});
+
+  @override
+  String toString() => {
+        'package': package,
+        'name': name,
+        'isDefault': isDefault,
+        'sources': sources,
+        'excludeSources': excludeSources,
+        'builders': builders
+      }.toString();
+}
+
+/// The configuration a particular [BuildTarget] applies to a Builder.
+///
+/// Build targets may have builders applied automatically based on
+/// [BuilderDefinition.autoApply] and may override with more specific
+/// configuration.
+class BuilderConfig {
+  /// Overrides the setting of whether the Builder would run on this target.
+  ///
+  /// Builders may run on this target by default based on the `apply_to`
+  /// argument. If this value is set it overrides the default.
+  final bool isEnabled;
+
+  /// Sources to use as inputs for this Builder in glob format.
+  ///
+  /// This is always a subset of the `include` argument in the containing
+  /// [BuildTarget].
+  ///
+  /// May be `null`, in which case it should fall back on `sources`.
+  final Iterable<String> generateFor;
+
+  final BuilderOptions options;
```

Comment:
lets add a comment here, mentioning it comes from the `options` argument

Reply:
Done